src/t_hash.c

```diff
@@ -134,6 +134,10 @@ hashTypeEntry *hashTypeCreateEntry(sds field, sds value) {
         /* Store the entry encoding type in sds aux bits. */
         sdsSetAuxBit(embedded_field_sds, FIELD_SDS_AUX_BIT_ENTRY_HAS_VALUE_PTR, 1);
         serverAssert(entryHasValuePtr(embedded_field_sds));
+        if (sdsType(value) == SDS_TYPE_32_SHARED) {
+            SDS_HDR_VAR(32shared, value);
+            sdsRetain(sh);
+        }
```

Comment:
This function has this doc comment:

    /* takes ownership of value, does not take ownership of field */

It means that we should *not* increment the reference counter. We take over the existing reference. The caller needs to increment it if they want to keep a reference to it.

```suggestion
```